### PR TITLE
Trigger onetime tests

### PR DESCRIPTION
**Pull request by rizakii**
test cases for https://github.com/KiiPlatform/thing-if-iOSSDK/pull/112

### CODE REVIEW

**Comment by rizakii** [comment]
@satoshikumano please review

File: ThingIFSDK/ThingIFSDK/ScheduleOncePredicate.swift [hunk 1]
```diff
@@ -25,12 +25,11 @@ public class ScheduleOncePredicate: NSObject,Predicate {
     }
 
     public required init(coder aDecoder: NSCoder) {
```
Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDK/ScheduleOncePredicate.swift:27]: NSDate is conforms to NSSecureCoding.
Why not just use NSDate capability for encoding/decoding?

File: ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift [hunk 2]
```diff
@@ -103,13 +111,22 @@ class PatchServerCodeTriggerTests: SmallTestBase {
...
+        let predicate = ScheduleOncePredicate(scheduleAt: NSDate(timeIntervalSinceNow: 1000))
+        patchServerCodeTrigger_http_404(predicate)
+    }
+    func patchServerCodeTrigger_http_404(predicate: Predicate) {
         let setting:TestSetting = TestSetting()
         let api = setting.api
         let tag = "PatchServerCodeTriggerTests.testPatchServerCodeTrigger_http_404"
         let triggerID = "0267251d9d60-1858-5e11-3dc3-00f3f0b5"
-        let expectation = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404")
+        weak var expectation : XCTestExpectation! = self.expectationWithDescription("testPostNewServerCodeTrigger_http_404_\(predicate.getEventSource().rawValue)")
```
Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift:129]: Can you add explanation about this change?

**Comment by satoshikumano** [comment]
@rizakii please confirm.

File: ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift (hunk 2 above)
Comment (rizakii) [inline ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift:129]: Please refer to https://github.com/KiiPlatform/thing-if-iOSSDK/pull/115
Defining expectations as let can caused execution timeout because of strong reference.
Refactoring another tests will be on another PR.

File: ThingIFSDK/ThingIFSDK/ScheduleOncePredicate.swift (hunk 1 above)
Comment (rizakii) [inline ThingIFSDK/ThingIFSDK/ScheduleOncePredicate.swift:27]: OK fixed.

File: ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift (hunk 2 above)
Comment (rizakii) [inline ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift:129]: please refer to https://github.com/apple/swift-corelibs-xctest/blob/eee815e96a068dded4b546e8693b65750ab552e9/Sources/XCTest/XCTestCase%2BAsynchronous.swift#L178
From the swift repo : 
- _allExpectations is an array which managed by xctest runtime
- Our testcases run parallel, in my opinion we should make all expectations `weak var`

additional information : 
- unlike regular class/subclass, testcase's `deinit` method is never called
- https://circleci.com/gh/KiiPlatform/thing-if-iOSSDK/356 shows that the test is timed out, this is absurd since it does not happen if run the test individually and the failure seems come from another unfullfilled expectations on another class.

Comment (satoshikumano) [inline ThingIFSDK/ThingIFSDKTests/PatchServerCodeTriggerTests.swift:129]: As far as i understood, people using weak attribute to avoid fulfill the expectation after the timeout.
But I don't think it's the best way since its hard to control when the expectation is nullified by ARC.
(In this point of view, I prefer to set nil in the completion handler explicitly instead of using weak)
I'm not very sure what was happend in the circleci build above but relation between adding weak and the failure is not very clear.

But OK. Let's separate the issue.
